frontend/www/js/omegaup/components/contest/Reportv2.vue

```diff
@@ -0,0 +1,174 @@
+            :fields="groupColumns"
+          >
+            <template #cell(details)="row">
+              <b-button
+                :title="T.wordsDetails"
+                variant="link"
+                @click="row.toggleDetails"
+              >
+                <b-icon-caret-down-square
+                  v-if="row.detailsShowing"
+                ></b-icon-caret-down-square>
+                <b-icon-caret-up-square v-else></b-icon-caret-up-square>
+              </b-button>
+            </template>
+            <template #row-details="row">
+              <b-table
+                responsive
+                striped
+                hover
+                :items="row.item.details"
+                :fields="casesColumns"
+              ></b-table>
+            </template>
+          </b-table>
+        </div>
+      </div>
+      <hr />
+    </b-card-text>
+  </b-card>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import T from '../../lang';
+import * as ui from '../../ui';
+
+// Import Bootstrap an BootstrapVue CSS files (order is important)
+import 'bootstrap/dist/css/bootstrap.css';
+import 'bootstrap-vue/dist/bootstrap-vue.css';
+
+// Import Only Required Plugins
+import {
+  TablePlugin,
+  ButtonPlugin,
+  CardPlugin,
+  BIconCaretDownSquare,
+  BIconCaretUpSquare,
+  BIconPrinter,
+} from 'bootstrap-vue';
+Vue.use(TablePlugin);
+Vue.use(ButtonPlugin);
+Vue.use(CardPlugin);
+
+interface GroupDetails {
+  name: string;
+  time: string;
+  wall_time: string;
+  memory: string;
+  verdict: string;
+  score: number;
+  diff?: string;
+}
+
+@Component({
+  components: {
+    BIconCaretDownSquare,
+    BIconCaretUpSquare,
+    BIconPrinter,
+  },
+})
+export default class Report extends Vue {
+  @Prop() contestReport!: types.ContestReport[];
+  @Prop() contestAlias!: string;
+
+  T = T;
+  ui = ui;
+  groupColumns = [
+    { key: 'group', label: T.wordsGroup, isRowHeader: true },
+    { key: 'rank_score', label: T.rankScore },
+    { key: 'details', label: T.wordsDetails },
+  ];
+
+  casesColumns = [
+    { key: 'name', label: T.wordsCase, isRowHeader: true },
+    { key: 'time', label: T.wordsTimeInSeconds },
+    { key: 'wall_time', label: T.wordsWallTimeInSeconds },
+    { key: 'memory', label: T.wordsMemoryInMebibytes },
+    { key: 'verdict', label: T.wordsStatus },
+    { key: 'score', label: T.rankScore },
+    { key: 'diff', label: T.wordsDifference },
+  ];
+
+  username(username: string): string {
+    return `${T.username}: ${username}`;
+  }
+
+  totalPoints(contestantData: types.ContestReport): string {
+    if (!this.hasTotalAndPointsProperties(contestantData)) {
+      return `${T.wordsTotal}: 0`;
+    }
+    return `${T.wordsTotal}: ${contestantData.total.points}`;
+  }
+
+  getGroupsByProblemAndUser(
+    groups: types.RunDetailsGroup[],
+  ): { group: string; rank_score: number; details: GroupDetails[] }[] {
```

Comment:
```suggestion
  ): { group: string; rankScore: number; details: GroupDetails[] }[] {
```

hay que intentar que todas las variables sean camelCase.

Reply:
Listo